tools/diagnostics/nsdiff

```diff
@@ -0,0 +1,660 @@
+#        The following were required for a python 3.5.2 installation:
+#
+#        `pip install pytest-runner python-rocksdb Cython rlp==0.6.0`
+#        TODO: Why does the setup.py in indy-plenum not correctly install
+#              pytest-runner, python-rocksdb, Cython, and rlp? At first glance,
+#              in indy-plenum/setup.py, the 'install-requires' section appears
+#              to contain python-rocksdb, and the 'setup-requires' section
+#              appears to contain pytest-runner. Cython is not explicitly
+#              declared anywhere in indy-plenum/setup.py. rlp does not designate
+#              a specific version and therefore defaults to the latest released
+#              version (currently >= 1.0.1). rlp version 0.6.0 is the latest
+#              version that appears to work with indy-plenum.
+#
+#        `pip install -e ${PWD}/indy-plenum`
+
+from __future__ import print_function
+from enum import Enum
+from ledger.compact_merkle_tree import CompactMerkleTree
+from ledger.hash_stores.file_hash_store import FileHashStore
+from ledger.ledger import Ledger
+from leveldb import LevelDBError
+from os import listdir
+from os.path import abspath, basename, dirname, isfile, isdir, join, splitext
+from plenum.common.constants import KeyValueStorageType
+from shutil import copyfile
+from storage.helper import initKeyValueStorage, initKeyValueStorageIntKeys
+from state.pruning_state import PruningState
+
+import argparse
+import json
+import os
+import shutil
+import subprocess
+import sys
+import tarfile
+import tempfile
+import zipfile
+
+class Error(Exception):
+    """Base class for exceptions in this module."""
+    pass
+
+class ArchiveError(Error):
+    """Exception raised for errors encounterd while identifying and extracting
+    tarfiles and zipfiles.
+
+    Attributes:
+        message -- explanation of the error
+    """
+    def __init__(self, message):
+        self.message = message
+
+# TODO: use standard python 'logging' module and map -v, -vv, etc. to logging
+#       log levels? STDERR can be the file to which the 'logging' module writes.
+#       Eliminates the need to check LOG_LEVEL (if >= LOG_LEVEL... logic)
+class LOG_LEVEL(Enum):
+    WARN = 1
+    INFO = 2
+    DEBUG = 3
+    TRACE = 4
+
+class NodeStateComparator():
+    verbosity = 0
+    cleanup = True
+    replay = False
+
+    s1 = "zipfile/tarfile"
+    s1dir = None
+    s1_is_temp = False
+    s1_transformed_and_scrubbed = None
+
+    s2 = "zipfile/tarfile"
+    s2dir = None
+    s2_is_temp = False
+    s2_transformed_and_scrubbed = None
+
+    def __init__(self, verbosity=0, cleanup=True, replay=False):
+        if self.verbosity >= LOG_LEVEL.DEBUG.value:
+            print("Initializing NodeStateComparator...")
+        self.verbosity = verbosity
+        self.cleanup = cleanup
+        self.replay = replay
+
+    def _cleanup(self):
+        if self.verbosity >= LOG_LEVEL.DEBUG.value:
```

Comment:
Initialising a `logger` in `__init__` of `NodeStateComparator` and using `logger.....` instead of `print` can be better.